### PR TITLE
Detect cleanups/v14

### PR DESCRIPTION
**Pull request by victorjulien**
replaces #10094, adding a few more things

https://redmine.openinfosecfoundation.org/issues/6661 (https://github.com/OISF/suricata/commit/a097d497323a5c178b075fda7b57066a7630023d)

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings. 

Pipeline 17352

**Review by inashivb — APPROVED** [review]
LGTM.

File: src/detect-pcre.h [hunk 1]
```diff
@@ -44,7 +44,6 @@ typedef struct DetectPcreData_ {
     DetectParseRegex parse_regex;
     int thread_ctx_id;
 
-    int opts;
```
Comment (catenacyber) [inline src/detect-pcre.h:47]: Did you find it with a compiler warning ?

File: src/detect.c [hunk 2]
```diff
@@ -1637,10 +1635,10 @@ static void DetectRunFrames(ThreadVars *tv, DetectEngineCtx *de_ctx, DetectEngin
 
             /* call individual rule inspection */
             RULE_PROFILING_START(p);
-            int r = DetectRunInspectRuleHeader(p, f, s, s->flags, s->proto.flags);
-            if (r == 1) {
+            bool r = DetectRunInspectRuleHeader(p, f, s, s->flags, s->proto.flags);
+            if (r == true) {
```
Comment (catenacyber) [inline src/detect.c:1639]: `if (r)` is good enough then ;-)

**Comment by victorjulien** [comment]
Merged in #10134, thanks!

**Pull request closed without merging**